prog/vm/nexus.rb

```diff
@@ -399,25 +411,61 @@ def write_params_json
 
     vm.update(display_state: "deleting")
 
+    clean_up_host
+    give_back_resources
+
+    hop_wait_lb_expiry if vm.load_balancer
+
+    final_clean_up
+
+    pop "vm deleted"
+  end
+
+  label def recreate
+    decr_recreate
+    clean_up_host(force_network_cleanup: true, omit_failure: true)
+    if vm.load_balancer
+      vm.load_balancer.evacuate_vm(vm)
+      vm.load_balancer.remove_vm(vm)
+    end
+
+    give_back_resources
+    vm.assigned_vm_address&.destroy
```

Comment:
funny thing, the predecessor of this program actually would hop to code shared with destroy (in fact, it was destroy) but be spared by "recreate" and then hop back to the provisioning path. Does doing something like that avoid repetition?

Reply:
Thanks for the suggestion, I restructured the code to use the `destroy` state which reduced the code footprint a low.